heathskin/game_state.py

```diff
@@ -18,40 +21,48 @@ def build_from_entities(cls, entities):
         gs.entities = entities
         return gs
 
-    def __init__(self):
+    def __init__(self, replay_from_log=False):
         self.logger = logging.getLogger()
-        self.start_new_game()
         self.players = {}
 
+        self.replay_from_log = replay_from_log
+
+        self.start_new_game()
+
         self.game_type = None
 
     def _create_history(self, *args, **kwargs):
         """ Create Game History after game Ends
         """
         history = GameHistory()
+        history.start_time = datetime.now()
         history.won = self._get_first_player_entity().get_tag('PLAYSTATE') == "WON"
+
+        history.end_time = datetime.now()
+
         if current_user:
             history.user_id = current_user.get_id()
         else:
             history.user_id = 0
         history.hero = kwargs.get('hero')
         history.opponent = kwargs.get('opponent')
-        history.enemy_health = 30
-        history.hero_health = 30
+        history.enemy_health, history.hero_health = self._get_player_healths()
         history.turns = kwargs.get('turns')
         history.first = not self._is_player_first()
         for player in self.players.values():
-          if player.get('first'):
-            history.player1 = player.get('username')
-          else:
-            history.player2 = player.get('username')
-        db.session.add(history)
-        db.session.commit()
+            if player.get('first'):
+                history.player1 = player.get('username')
+            else:
+                history.player2 = player.get('username')
+
+        if not self.replay_from_log:
```

Comment:
move this back inside game_state: don't even call `_create_history` if `self.replay_from_log` is set